dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -1739,57 +1739,62 @@ Sedp::disassociate(ParticipantData_t& pdata)
     remove_from_bit_i(*it);
   }
 
+  BuiltinEndpointSet_t associated_endpoints = pdata.associated_endpoints;
```

Comment:
```suggestion
  BuiltinEndpointSet_t& associated_endpoints = pdata.associated_endpoints;
```

Reply:
These can't be references, because pdata can be invalidated while the reverse lock is in effect. See the end, where I write this value back into pdata (if it exists).